101-iothub-device-provisioning/azuredeploy.json

```diff
@@ -62,7 +62,7 @@
                 "capacity": "[parameters('skuUnits')]"
             },
             "properties": {
-                "minTlsVersion": "[if(equals(environment().name, 'AzureUSGoverment'), parameters('minTLS'), json('null'))]"
+                "minTlsVersion": "[if(equals(environment().name, 'AzureUSGovernment'), parameters('minTLS'), json('null'))]"
```

Comment:
@bmoore-msft This works but, TLS 1.2 is only required for `usgovvirginia`. It is **not** required for `usgovarizona`.

Pretty sure `AzureUSGovernment` is comprised of both  `usgovvirginia` and `usgovarizona`.  Seems like security requirements would be the same across both but it's not.

Reply:
Ok, that's a little strange... but if those are the rules, I'm inclined to say we should just doc it, instead of build it into the template.  Thoughts?

Also thinking about the scenario where I want to set the min tls version but am running in another cloud/region...

Reply:
I definitely have to document this.  But, that pipeline failure is a irritating to me.  My vote would be for just leaving `usgovvirginia` specified since that is a definite requirement. This way its clear in the docs and the template

Reply:
We try to avoid whenever possible hard-coding locations in the templates.  Eventually... that will cause a failure when it shouldn't...  I took a look at the README and it's not clear to me what's needed.  I took a look at the [current] doc linked to in the README and it doesn't mention anything there either...

Soooo.... Here's what I think you're saying the requirement is...  

> Only TLS 1.2 will work in conjuction with DPS from other clouds/regions. iot Hubs still supports TLS 1.0 and 1.1 but DPS does not.

When DPS is provisioned, TLS 1.2 must be used.  This template unconditionally deploys DPS, so TLS 1.2 would always be required for this template.  If this is true we can just hard-code TLS minVersion to 1.2.

If, OTOH, it's only ```usgovvirginia``` that has this requirement, we should state that in the docs and let the use select the TLS version since the template may not be deployed to ```usgovvirginia``` or the use may still want to use 1.2 even though it's not required.

Am I following this correctly?

Reply:
No. A few clarifications:

* DPS supports using TLS (and recommends enabling it) to restrict device access but it's not required
* If TLS is enabled, DPS only supports TLS 1.2
* IoT Hubs support TLS (and recommends enabling it) to restrict device access but it's not required except in `usgovvirginina`
* IoT Hubs support TLS 1.0, 1.1, and 1.2

So, TLS will always be required when deploying IoT hubs in `usgovvirginia`. If DPS is being used with the hub then TLS 1.2 must be used.

I understand that we would want to avoid hard-coding a location. However, I would think this would be a good exception as it would help clarifiy and reinforce to the customer that this is not optional. As a starter template in the hands of a customer, I'd think customer's would appreciate a template that helps reinforce this without them having to run into problems with this. 

I'll update the doc today to reflect this. Let me know what you decide for this repo and I'll fix the PR

Reply:
One slight correction to the above. Based on the information I have from the PM, IoT hubs in `usgovvirginia` must support TLS 1.2. 

So no TLS 1.0 or 1.1 support in `usgovvirginia` location

Reply:
Ok, thanks for the matrix...  I had a bit of a chuckle

Here's what I think we should do:
1. template should require TLS 1.0, 1.1, 1.2 - I know IoT doesn't require it but we should since it's recommended.  If someone doesn't want TLS they can use a different sample or modify the template
1. user selects a TLS version via param
1. if location is usgovvirgina (sigh...) change the param value to 1.2
1. doc all of this in the README.MD - even the nit that TLS isn't required but this sample uses it

I can make the changes to the template if you want to make the doc changes (or you can do both if that's easier)

Reply:
Sorry, I just seen this.  I don't want to mention TLS 1.0 and 1.1 in this template because this is a DPS quickstart template.  Things will not work if they go that route.

So I just set a parameter to enable TLS